### PR TITLE
Fix postgres nodes assertion

### PR DESCRIPTION
**Pull request by seriva**

### CODE REVIEW

File: cli/engine/ApplyEngine.py [hunk 1]
```diff
@@ -123,25 +124,30 @@ def assert_no_master_downscale(self):
...
-                next_postgres_count = int(components['postgresql']['count'])
-
-                if prev_postgres_count != next_postgres_count:
-                    raise Exception("Postgresql scaling is not supported yet. Please revert your 'postgresql' count to previous value.")
+            prev_postgres_node_count = len(existing_inventory.list_hosts(pattern='postgresql'))
+            postgres_available = [x for x in feature_mapping.specification.available_roles if x.name == 'postgresql']
+            if postgres_available[0].enabled:
+                for key, roles in feature_mapping.specification.roles_mapping.items():
+                    if ('postgresql') in roles:
+                        if key in components:
```
Comment (atsikham) [inline cli/engine/ApplyEngine.py]: May be combine them?

**Review by atsikham — CHANGES_REQUESTED** [review]

**Review by plirglo — DISMISSED** [review]

File: cli/engine/ApplyEngine.py (hunk 1 above)
Comment (seriva) [reply cli/engine/ApplyEngine.py]: Good point.

**Review by plirglo — APPROVED** [review]

**Review by atsikham — APPROVED** [review]

**Comment by atsikham** [comment]
✅  Checked different combinations of scaling up/down and feature mapping changes for apply. Everything works as expected - there is an error message in case of PostgreSQL nodes number change.

**Comment by przemyslavic** [comment]
@seriva I think it can be merged as it has already been tested by @atsikham.

**Pull request merged**